src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,529 @@
+        &const_interp_trajectory, &trajectory, 0.001);
+
+    EXPECT_NEAR(trajectory.trajectory_elements[trajectory.num_elements - 1].position.x,
+                const_interp_trajectory
+                    .trajectory_elements[const_interp_trajectory.num_elements - 1]
+                    .position.x,
+                0.001);
+    EXPECT_NEAR(trajectory.trajectory_elements[trajectory.num_elements - 1].position.y,
+                const_interp_trajectory
+                    .trajectory_elements[const_interp_trajectory.num_elements - 1]
+                    .position.y,
+                0.001);
+    EXPECT_NEAR(trajectory.trajectory_elements[trajectory.num_elements - 1].time,
+                const_interp_trajectory
+                    .trajectory_elements[const_interp_trajectory.num_elements - 1]
+                    .time,
+                0.001);
+
+    EXPECT_NEAR(trajectory.trajectory_elements[0].position.x,
+                const_interp_trajectory.trajectory_elements[0].position.x, 0.001);
+    EXPECT_NEAR(trajectory.trajectory_elements[0].position.y,
+                const_interp_trajectory.trajectory_elements[0].position.x, 0.001);
+    EXPECT_NEAR(trajectory.trajectory_elements[0].time,
+                const_interp_trajectory.trajectory_elements[0].time, 0.001);
+}
+
+TEST_F(TrajectoryPlannerTest, test_assert_cannot_reach_final_velocity)
+{
+    Polynomial2dOrder3_t path = {
+        .x = {.coefficients = {0, 0, 1, 0}},
+        .y = {.coefficients = {0, 0, 1, 0}},
+
+    };
+
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.path                       = path;
+    path_parameters.t_start                    = 0;
+    path_parameters.t_end                      = 1;
+    path_parameters.num_segments               = 5999;
+    path_parameters.max_allowable_acceleration = 3;
+    path_parameters.max_allowable_speed        = 2000;
+    path_parameters.initial_speed              = 0;
+    path_parameters.final_speed                = 3000;
+
+    TrajectoryElement_t const_arc_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    Trajectory_t trajectory = {.trajectory_elements = const_arc_elements};
+
+
+    enum TrajectoryPlannerGenerationStatus status =
+        app_trajectory_planner_generateConstantArcLengthSegmentation(path_parameters,
+                                                                     &trajectory);
+
+    EXPECT_EQ(finalVelocityTooHigh, status);
+}
+
+TEST_F(TrajectoryPlannerTest, test_assert_initial_velocity_too_high)
+{
+    Polynomial2dOrder3_t path = {
+        .x = {.coefficients = {3, 0, 1, 0}},
+        .y = {.coefficients = {2, 0, 1, 0}},
+
+    };
+
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.path                       = path;
+    path_parameters.t_start                    = 0;
+    path_parameters.t_end                      = 10;
+    path_parameters.num_segments               = 5999;
```

Comment:
Seems like in several tests you use `5999` as the `num_segments`, which seems like a hardcoded number to be just below `TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS` from back when it was `6000` (IIRC it was 6000 at one point).

I'm relatively confident if we ever decreased `TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS` below this  number for whatever reason the tests would fail (compiler may catch the allocation issue, or we have asserts.

However if for these tests we really are just testing the highest fidelity path possible (most points in the trajectory allowable), maybe it's reasonable to just use `TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS` directly here? I'm generally not ok with using constants directly in tests, but it may be reasonable in this case along with a quick comment like "Exact num segments don't matter so we use the max allowable"

@garethellis0 ?

Reply:
Yeah, I think using `TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS-1` here would be fine.

Reply:
I changed lots of the tests and added some comments to the num_segments